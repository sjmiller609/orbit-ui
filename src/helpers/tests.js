import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
