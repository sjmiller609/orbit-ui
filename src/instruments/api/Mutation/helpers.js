const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
