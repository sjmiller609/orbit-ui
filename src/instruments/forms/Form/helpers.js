// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
