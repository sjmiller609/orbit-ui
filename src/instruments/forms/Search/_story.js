import React from 'react'
import { storiesOf } from '@storybook/react'
import { Search } from 'instruments'

storiesOf('Instruments|Forms.Search', module)
  .add('Default', () => (
    <Search
      search={() => null}
      text="Search"
      placeholder="Type to search"
      updateErrors={() => null}
    />
  ))
  .add('Dark', () => (
    <Search
      search={() => null}
      text="Search"
      placeholder="Type to search"
      updateErrors={() => null}
      dark={true}
    />
  ))
