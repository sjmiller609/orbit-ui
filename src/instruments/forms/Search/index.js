'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import s from './styles.scss'
// NOTE: Importing field styles from TextField
import field from '../Field/styles.scss'
import classnames from 'classnames'

import { LoadingDots } from 'instruments'

class Search extends React.Component {
  timeout = null
  type = this.type.bind(this)
  clear = this.clear.bind(this)
  state = {
    typing: null,
    search: this.props.text,
  }

  componentWillUnmount() {
    clearTimeout(this.timeout)
  }

  type(e) {
    e.preventDefault()
    const { noDelay, search } = this.props
    const text = e.target.value
    this.setState({ search: text })

    if (noDelay) search(e.target.value)
    else {
      // put 1 second delay on actual search from when stop typing
      if (this.timeout) clearTimeout(this.timeout)
      this.timeout = setTimeout(() => {
        this.props.search(text)
        clearTimeout(this.timeout)
      }, 1000)
    }
  }

  clear(e) {
    e.preventDefault()

    this.setState({ search: '' })
    this.props.search('')
  }

  render() {
    const { placeholder, text, dark, className } = this.props
    const { search } = this.state

    return (
      <div
        className={classnames(
          field.field,
          s.search,
          className,
          dark ? s.dark : null
        )}>
        <input
          type="text"
          placeholder={placeholder}
          value={search}
          onChange={this.type}
          className={s.input}
        />
        {search !== text ? (
          <div className={s.searching}>
            <LoadingDots />
          </div>
        ) : (
          text && (
            <div className={s.clear} onClick={this.clear}>
              x
            </div>
          )
        )}
      </div>
    )
  }
}

Search.propTypes = {
  search: PropTypes.func.isRequired,
  text: PropTypes.string,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  dark: PropTypes.bool,
  noDelay: PropTypes.bool,
}

Search.defaultProps = {
  text: '',
  placeholder: 'Search',
  dark: false,
}

export default Search
