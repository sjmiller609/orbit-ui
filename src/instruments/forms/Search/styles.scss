@import '../../../styles/mixins.scss';

.search {
  position: relative;
  // inherits vertical padding from form
  //max-width: $fieldWidth;
  width: 100%;

  &:after {
    content: '';
    position: absolute;
    top: 0;
    left: 0.5rem;
    z-index: 1;
    background: url('./search.svg') no-repeat center center;
    background-size: 1.2rem 1.2rem;
    width: 1.2rem;
    height: 100%;
  }

  .input {
    margin: 0;
    width: 100%;
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }

  .searching {
    position: absolute;
    top: 0;
    right: 0.5rem;
    z-index: 1;
    font-size: 3rem;
    line-height: 1rem;
    span {
      color: transparent;
      background-clip: text;
      -webkit-background-clip: text;
      background-image: linear-gradient(to right, $green, $blue);
    }
  }

  &.dark {
    input {
      color: $darkWhite;
      background-color: $purple;
      &:hover,
      &:focus {
        background-color: $purple;
      }
    }
  }
}
