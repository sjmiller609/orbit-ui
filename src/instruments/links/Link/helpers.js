// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
