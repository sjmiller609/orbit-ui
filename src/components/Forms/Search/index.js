import React, { Component } from 'react';

import Form from 'components/Form';
import TextInput from 'components/Form/TextInput';
import Search from 'components/Icons/Search';

import styles from './styles.module.css';

class SearchForm extends Component {
  state = {
    payload: {
      search: '',
    },
  }

  handleChange = (e, isValid) => {
    this.setState({
      payload: {
        ...this.state.payload,
        [e.target.name]: e.target.value,
      },
    });
  }

  render() {
    const { handleSubmit, error, placeholder } = this.props;
    const { payload } = this.state;

    return (
      <div className={styles.container}>
        <Form handleSubmit={e => handleSubmit(payload, e)} error={error}>
          <span>
            <Search />
          </span>
          <TextInput
            name="search"
            placeholder={placeholder}
            type="text"
            value={payload.search}
            handleChange={this.handleChange}
            required
          />
        </Form>
      </div>
    );
  }
}

export default SearchForm;
