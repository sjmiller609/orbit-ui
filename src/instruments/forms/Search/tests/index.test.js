import React from 'react'
import { mountWrap } from 'helpers/tests'
import Search from '../index'

const props = {
  search: jest.fn(),
  text: 'Test',
  placeholder: 'Search',
  className: 'testClassString',
  dark: false,
  noDelay: true,
}

describe('Search', () => {
  it('renders correctly', () => {
    const wrapper = mountWrap(<Search {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
