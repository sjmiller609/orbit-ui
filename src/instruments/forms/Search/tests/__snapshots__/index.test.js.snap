// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<Search
  className="testClassString"
  dark={false}
  noDelay={true}
  placeholder="Search"
  search={[Function]}
  text="Test"
>
  <div
    className="field search testClassString"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="Test"
    />
    <div
      className="clear"
      onClick={[Function]}
    >
      x
    </div>
  </div>
</Search>
`;
